tests/annotations/tests.py

```diff
@@ -148,6 +148,16 @@ def test_mixed_type_annotation_numbers(self):
         combined = int(test.pages + test.rating)
         self.assertEqual(b.combined, combined)
 
+    def test_empty_expression_annotation(self):
+        books = Book.objects.annotate(
+            selected=ExpressionWrapper(Q(pk__in=[]), output_field=BooleanField())
+        )
+        self.assertGreater(len(books), 0)
```

Comment:
Why such a vague assertion? At least you could run `Book.objects.count()` at the beginning of the test and check the count if that's the expected result. What about also verifying the purpose of the annotation -- it's not obvious to me what that purpose is.